### DISCUSSION TITLE
What are production quality router options for the `mobx-state-tree`-based web-apps?

### DISCUSSION THREAD

**Discussion by SergeyVolynkin** [mobxjs/mobx-state-tree#1721] [Q&A] [answered]
Greetings,

I've searched over the internet and didn't get a good understanding / didn't find good suggestions on what routing library should be a good fit for the `mobx-state-tree`-based web-apps

Maybe someone from the MST community can share their production experience?

**Comment by SergeyVolynkin** [comment]
My thoughts that 
1. [kitze/mobx-router](https://github.com/kitze/mobx-router) Is an interesting candidate
   - Cons: 
        - there is lack of documentation on how to use [kitze/mobx-router](https://github.com/kitze/mobx-router) + [mobx-state-tree ](https://github.com/mobxjs/mobx-state-tree) together
2. [react-router](https://github.com/ReactTraining/react-router) should be a robust solution that most of the FE devs know & used before
   - Cons:
        - As far as I'm concerned it's not easy (possible?) to separate data-fetching from component lifecycle methods (e.g. like [kitze/mobx-router](https://github.com/kitze/mobx-router) allows with `beforeEnter: () => {…}`)
        
        
cc @mweststrate @kitze @ryanflorence would really appreciate your opinion if you have a few min.

**Reply by kitze** [reply]
@SergeyVolynkin check out [rttr](https://github.com/kitze/rttr) (not maintained btw)

**Comment by mweststrate** [comment, marked as answer]
See also https://nareshbhatia.github.io/mobx-state-router/ /
https://github.com/miles-till/mobx-state-tree-router. No personal
experience with any of them.

**Reply by SergeyVolynkin** [reply]
Thanks, @mweststrate! I'll look into these

If it's okay to ask, what do you tend to use in your prod projects for routing? (together with `mobx`/`mobx-state-tree`)

**Comment by SergeyVolynkin** [comment]
@nareshbhatia if you have a few minutes, may I ask for your recommendations on the topic?

**Reply by nareshbhatia** [reply]
Hi @SergeyVolynkin, I am not experienced in mobx-state-tree, so don't know if mobx-state-router will work with it or not. All I can say is that the the router is well documented (see [here](https://nareshbhatia.github.io/mobx-state-router/)). Please feel free to try it and let me know if it works.

P.S. Note that mobx-state-router version 6.0.0-beta.2 supports mobx-6

**Reply by SergeyVolynkin** [reply]
Ack, thank you @nareshbhatia

**Comment by EmilTholin** [comment]
Michel's blogpost [How to decouple state and UI](https://medium.com/hackernoon/how-to-decouple-state-and-ui-a-k-a-you-dont-need-componentwillmount-cc90b787aa37) is still a favorite of mine all these years later. It outlines how to invert the concern you have with React Router with vanilla MobX, but it's straightforward to migrate the concepts to MST.

**Comment by paxarpp** [comment]
Hi, i used https://github.com/alisd23/mst-react-router
